File: TAKAlertSample/TAKAlertSample/ViewController.m
//
//  ViewController.m
//  TAKAlertSample
//
//  Created by 西村 拓 on 2015/07/27.
//  Copyright (c) 2015年 TakuNishimura. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (nonatomic) int selectedIndex;

@property (weak, nonatomic) IBOutlet UILabel *resultLabel;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)showButtonTapped:(UIButton *)sender {
    __weak typeof(self) weakSelf = self;
    [TAKAlertUtil showWithTitle:@"Title"
                        message:@"Message Test"
                   buttonTitles:@[@"Cancel",@"OK"]
                  buttonHandler:^(TAKAlertUtil *alertView, NSInteger index) {
                      if (index > 0) {
                          weakSelf.resultLabel.text = @"OK";
                      } else {
                          weakSelf.resultLabel.text = @"Cancel";
                      }
                  }];
}

@end

File: ExampleSwift/TAKAlertSample/ViewController.swift
//
//  ViewController.swift
//  TAKAlertSample
//
//  Created by 西村 拓 on 2015/09/16.
//  Copyright © 2015年 TakuNishimura. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var label: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func alertButtonTapped(sender: UIButton) {
        TAKAlertView.show(
            title: "Title",
            message: "Sample message.",
            buttonTitles: ["OK", "Cancel"]
        ).subscribeNext { index -> Void in
            print(index)
        }
    }
}
